types/utils.go

```diff
@@ -58,3 +66,17 @@ func ParseTimeBytes(bz []byte) (time.Time, error) {
 	}
 	return t.UTC().Round(0), nil
 }
+
+// NewLevelDB instantiate a new LevelDB instance according to DBBackend.
+func NewLevelDB(name, dir string) (db dbm.DB, err error) {
+	backend := dbm.GoLevelDBBackend
+	if DBBackend == string(dbm.CLevelDBBackend) {
+		backend = dbm.CLevelDBBackend
+	}
+	defer func() {
+		if r := recover(); r != nil {
```

Comment:
Does the underlying `NewDB` call panic instead of returning an error we can catch?

Reply:
It unfortunately does, yes

Reply:
OK, this is fine then.